### PR TITLE
Silvestreperret/add cs runner

### PR DESCRIPTION
**Pull request by SilvestrePerret**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -95,6 +95,11 @@ jobs:
       #   if: steps.cache-cargo.outputs.cache-hit != 'true'
       #   run: curl https://sh.rustup.rs -sSf -o rustup.sh && sh rustup.sh -y
 
+
```
Comment (badouralix) [inline .github/workflows/main.yml:98]:
```suggestion
```

File: tool/runners/cs.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import tempfile
+
+from tool.runners.exceptions import CompilationError, RuntimeError
+from tool.runners.wrapper import SubmissionWrapper
+
+
+class SubmissionCs(SubmissionWrapper):
+    def __init__(self, file):
+        SubmissionWrapper.__init__(self)
+        # Create a temporary directory to put the compiled java in,
```
Comment (badouralix) [inline tool/runners/cs.py:13]:
```suggestion
        # Create a temporary directory to put the compiled C# in,
```

File: tool/runners/cs.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                # executable not found
+                raise CompilationError(e)
+            else:
+                # subprocess exited with another error
+                raise RuntimeError(e)
+
+    def __call__(self):
+        return SubmissionCs(self.file)
```
Comment (badouralix) [inline tool/runners/cs.py:51]: Trying to fix the no newline at end of file ( not sure it works, maybe you need to make the change locally )

```suggestion
        return SubmissionCs(self.file)
```

File: tool/templates/template.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            string input = args[0];
+            var watch = new System.Diagnostics.Stopwatch();
+            watch.Start();
+            string result = solve(input);
+            watch.Stop();
+            Console.WriteLine("_duration: " + watch.ElapsedMilliseconds + "\n" + result);
+        }
+        
+    }
+}
```
Comment (badouralix) [inline tool/templates/template.cs:24]: Trying to fix the no newline at end of file ( not sure it works, maybe you need to make the change locally )

```suggestion
}
```

**Review by badouralix — APPROVED** [review]
Added a few nitpicks

As long as it does roughly what we need, lgtm :shipit:
And we can still fix things in the future if ever we discover a new pitfall

**Comment by SilvestrePerret** [comment]
@badouralix 
I will wait until some submissions are available to make sure everything run smoothly on the CI.
I will rebase my submission branches on this one until it's merged.

**Pull request merged**